test/nodes/test_connector.py

```diff
@@ -184,3 +185,24 @@ def test_crawler_loading_wait_time(test_url, tmp_path):
     assert content_in_results(crawler, test_url + "/index.html", paths)
     assert content_in_results(crawler, test_url + "/page1.html", paths)
     assert content_in_results(crawler, test_url + "/page2.html", paths)
+
+
+def test_crawler_naming_function(test_url, tmp_path):
+    crawler = Crawler(output_dir=tmp_path, crawler_naming_function=lambda link, text: link)
+
+    link = f"{test_url}/page_dynamic.html"
+    link_split_values = (
+        link.replace("https://", "")
+        .replace("http://", "")
+        .replace("file:/", "")
+        .replace("file://", "")
+        .replace("\0", "")
+        .split("/")
+    )
+    file_name = f"{'_'.join(link_split_values)}.json"
+    expected_crawled_file_path = tmp_path / file_name
```

Comment:
It would be, but crawling local files without removing forbidden chars would never save the file. So I had to implement it on crawler, and then on the test.